pkgs/development/python-modules/pydruid/default.nix

```
@@ -0,0 +1,69 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+  # required dependencies
+, requests
+, setuptools
+  # optional dependencies
+, pandas
+, tornado
+, sqlalchemy
+, pygments
+, prompt-toolkit
+, tabulate
+  # test dependencies
+, pycurl
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "pydruid";
+  version = "0.6.5";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    repo = pname;
+    owner = "druid-io";
+    rev = "refs/tags/${version}";
+    hash = "sha256-9+xomjSwWDVHkret/mqAZKWOPFRMvVB3CWtFPzrT81k=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  propagatedBuildInputs = [
+    requests
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    sqlalchemy
+    tornado
+    pandas
+    pycurl
+  ];
```

```suggestion
  nativeCheckInputs = [
    pytestCheckHook
    pycurl
  ] ++ lib.flatten (lib.attrValues passthru.optional-dependencies);
```

At some point it also makes sense to include all optional-dependencies in `nativeCheckInputs`. Not saying this is the point, given that you currently don't pass the `cli` extra deps, but it is helpful to keep this expression in mind.

---

Roger that. I just included optional dependencies until the tests passed.

---

In which case you should also watch out for skipped tests.

---

Indeed.

In this case there are no skipped tests:

```
python3.10-pydruid> ============================= test session starts ==============================
python3.10-pydruid> platform linux -- Python 3.10.10, pytest-7.2.1, pluggy-1.0.0
python3.10-pydruid> rootdir: /build/source
python3.10-pydruid> collected 108 items
python3.10-pydruid> tests/test_async_client.py .....                                         [  4%]
python3.10-pydruid> tests/test_client.py .....                                               [  9%]
python3.10-pydruid> tests/test_query.py ........                                             [ 16%]
python3.10-pydruid> tests/db/test_cursor.py .......                                          [ 23%]
python3.10-pydruid> tests/db/test_druid_dialect.py ....                                      [ 26%]
python3.10-pydruid> tests/db/test_rows_from_chunks.py .....                                  [ 31%]
python3.10-pydruid> tests/utils/test_aggregators.py .....                                    [ 36%]
python3.10-pydruid> tests/utils/test_dimensions.py ..........................                [ 60%]
python3.10-pydruid> tests/utils/test_filters.py ............................                 [ 86%]
python3.10-pydruid> tests/utils/test_having.py .............                                 [ 98%]
python3.10-pydruid> tests/utils/test_query_utils.py ..                                       [100%]
python3.10-pydruid> ============================= 108 passed in 0.70s ==============================
```